common/database/mongo_database.go

```diff
@@ -205,6 +206,20 @@ func (db *MongoDatabase) UpdateAll(collection_name string, selector interface{},
 	return &change_results, convertMgoError(err)
 }
 
+/*
+	Finds an item based on the given selector and patches it with the data in update
+*/
+func (db *MongoDatabase) Patch(collection_name string, selector interface{}, update *map[string]interface{}) error {
```

Comment:
If it's not a lot of extra effort, it would be nice to have `update` be an `interface{}` to match the signature of the other database functions. Not a big deal if not.